cli/reactjs_jsx_ppx_v3.ml

```diff
@@ -317,953 +317,935 @@ let makePropsRecordTypeSig propsName loc namedTypeList =
-    (* is it a list with jsx attribute? Reason <>foo</> desugars to [@JSX][foo]*)
-    | {
-        pexp_desc =
-          ( Pexp_construct
-              ({txt = Lident "::"; loc}, Some {pexp_desc = Pexp_tuple _})
-          | Pexp_construct ({txt = Lident "[]"; loc}, None) );
-        pexp_attributes;
-      } as listItems -> (
-      let jsxAttribute, nonJSXAttributes =
-        List.partition
-          (fun (attribute, _) -> attribute.txt = "JSX")
-          pexp_attributes
+      let childrenExpr = transformChildrenIfList ~loc ~mapper listItems in
+      let args =
+        [
+          (* "div" *)
+          (nolabel, fragment);
+          (* [|moreCreateElementCallsHere|] *)
+          (nolabel, childrenExpr);
+        ]
       in
-      match (jsxAttribute, nonJSXAttributes) with
-      (* no JSX attribute *)
-      | [], _ -> default_mapper.expr mapper expression
-      | _, nonJSXAttributes ->
-        let loc = {loc with loc_ghost = true} in
-        let fragment =
-          Exp.ident ~loc {loc; txt = Ldot (Lident "ReasonReact", "fragment")}
-        in
-        let childrenExpr = transformChildrenIfList ~loc ~mapper listItems in
-        let args =
-          [
-            (* "div" *)
-            (nolabel, fragment);
-            (* [|moreCreateElementCallsHere|] *)
-            (nolabel, childrenExpr);
-          ]
-        in
-        Exp.apply
-          ~loc (* throw away the [@JSX] attribute and keep the others, if any *)
-          ~attrs:nonJSXAttributes
-          (* ReactDOMRe.createElement *)
-          (Exp.ident ~loc
-             {loc; txt = Ldot (Lident "ReactDOMRe", "createElement")})
-          args)
-    (* Delegate to the default mapper, a deep identity traversal *)
-    | e -> default_mapper.expr mapper e
-    [@@raises Invalid_argument]
-  in
+      Exp.apply
+        ~loc (* throw away the [@JSX] attribute and keep the others, if any *)
+        ~attrs:nonJSXAttributes
+        (* ReactDOMRe.createElement *)
+        (Exp.ident ~loc
+           {loc; txt = Ldot (Lident "ReactDOMRe", "createElement")})
+        args)
+  (* Delegate to the default mapper, a deep identity traversal *)
+  | e -> default_mapper.expr mapper e
+  [@@raises Invalid_argument]
 
-  let module_binding mapper module_binding =
-    let _ = nestedModules := module_binding.pmb_name.txt :: !nestedModules in
-    let mapped = default_mapper.module_binding mapper module_binding in
-    let _ = nestedModules := List.tl !nestedModules in
-    mapped
-    [@@raises Failure]
-  in
+let module_binding nestedModules mapper module_binding =
+  let _ = nestedModules := module_binding.pmb_name.txt :: !nestedModules in
+  let mapped = default_mapper.module_binding mapper module_binding in
+  let _ = nestedModules := List.tl !nestedModules in
+  mapped
+  [@@raises Failure]
+
+(* TODO: some line number might still be wrong *)
+let jsxMapper nestedModules =
+  let jsxVersion = ref None in
```

Comment:
I've remained the `jsxVersion` ref still, anyway what is it for? I've looked into the compiler code base too, but it seems not to have a specific job.

Reply:
Sounds good.